### DISCUSSION TITLE
Why does this simple SurfaceTexture never become available in MAUI, but it does in Forms?

### DISCUSSION THREAD

**Discussion by KRA2008** [dotnet/maui#19705] [Q&A]
In my big project I'm using a custom renderer for a ContentView so that I can use a TextureView to interact with the camera. After migrating to MAUI I find that the SurfaceTexture never becomes available. I recreated an extremely simple version of this problem in a Forms project and a MAUI project. I find that the SurfaceTexture becomes available in Forms but it doesn't in MAUI. Is it a bug? Am I missing something?

Just run the app and you'll see that in the Forms version we hit the Debugger.Break() in the SurfaceTextureAvailable callback, and in the MAUI we do not.

- https://github.com/KRA2008/ContentViewRendererForms.git

- https://github.com/KRA2008/ContentViewRendererMaui.git

**Comment by tmijieux** [comment]
i had a similar problem and in my case the texture became available on the first time i install the app and the app show the dialog to ask for permission, but every other time the texture did not became available. 

So i understood  that this was a timing issue, when i already had the permission the ui was trying to get the texture immediatly  (probably before the view in maui world was set up with a valid size ??)  and the end result no texture became available maybe because it was asking a texture of size 0,0 ? (i think something like that was happening) .
So i did a check if i already has permission (before requesting immediately) and in that case I add a 1 second delay before creating my native view and asking for a texture and it worked.

Hope this can be of some help to you.

**Reply by tagos777** [reply]
can you send me a code where did you exactly added a delay please?

**Reply by tmijieux** [reply]
> can you send me a code where did you exactly added a delay please?


https://github.com/tmijieux/MinimalCameraAppMAUI/

here is what you asked specifically
https://github.com/tmijieux/MinimalCameraAppMAUI/blob/3e22ba66c234ae35b1cf7ce84e48837fe58b77e6/Platforms/Android/Renderers/Camera/CameraViewRenderer.cs#L110-L123

edit: note the wasRequired variable here means that we did not have the permission before calling the RequestPermissionIfRequired() method.


so we had the delay in the case where wasRequired is false meaning IN THE CASE we already had the given the permission in a previous usage of the app
